driver/pace/driver/state.py

```diff
@@ -175,38 +194,160 @@ def _overwrite_state_from_restart(
     return state
 
 
+def _overwrite_state_from_fortran_restart(
+    path: str,
+    communicator: pace.util.CubedSphereCommunicator,
+    state: Union[fv3core.DycoreState, pace.physics.PhysicsState, TendencyState],
+    is_gpu_backend: bool,
+):
+    """
+    Args:
+        path: path to restart files
+        rank: current rank number
+        state: an empty state
+        restart_file_prefix: file prefix name to read
+
+    Returns:
+        state: new state filled with restart files
+    """
+    state_dict = _driver_state_to_dict(state)
+
+    state_dict = pace.util.open_restart(path, communicator)
+
+    state = _dict_state_to_driver_state(state_dict, state, is_gpu_backend)
+
+    return state
+
+
+def _driver_state_to_dict(
+    driver_state: Union[fv3core.DycoreState, pace.physics.PhysicsState, TendencyState],
+):
+    """
+    Takes a Pace driver state
+    and returns a dict of state quantities with their Fortran names
+    """
+
+    dict_state = {}
+
+    for field in fortran_restart_to_pace_dict.keys():
+        dict_state[fortran_restart_to_pace_dict[field]] = driver_state.__dict__[
+            field
+        ].data[:]
+
+    # Ajda
+    # at this point the data for every field in dict_state is 0.
+    return dict_state
+
+
+def _dict_state_to_driver_state(
+    fortran_state: dict,
+    driver_state: Union[fv3core.DycoreState, pace.physics.PhysicsState, TendencyState],
+    is_gpu_backend: bool,
+):
+    """
+    Takes a dict of state quantities with their Fortran names and a driver state
+    and populates the driver state with quantities from the dict.
+    """
+
+    for field in fortran_restart_to_pace_dict.keys():
+        driver_state.__dict__[field].view[:] = np.transpose(fortran_state[field].data)
+
+        if is_gpu_backend:
+            # driver_state.__dict__[field].view[:] = gt_utils.asarray(
+            #     np.transpose(fortran_state[field].data), to_type=cp.ndarray,
+            # )
+            # Ajda
+            # not sure if this will work?? Internet told me cupy has transpose
```

Comment:
Is there any way to find out before merging? @elynnwu we might want to add GPU-backend coverage of or non-savepoint (main) tests.